setup-py-to-pyproject-toml.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+# This script will convert setup.py to pyproject.toml
+# for ITK remote modules. It will also update the
+# itk version to 5.4.* in the dependencies section.
+
+import os
+import re
+import sys
+import argparse
+
+def setup_py_to_pyproject_toml(setup_python_path):
+    if not os.path.exists(setup_python_path):
+        print('setup.py file not found')
+        sys.exit(1)
+
+    with open(setup_python_path, 'r') as f:
+        setup_py = f.read()
+
+    project_name = re.search(r'name\s*=\s*[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_version = re.search(r'version\s*=\s*[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_description = re.search(r'description\s*=\s*[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_author = re.search(r'author\s*=\s*[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_author_email = re.search(r'author_email\s*=\s*[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_url = re.search(r'url\s*=\s*r?[\'"]([^\'"]*)[\'"]', setup_py).group(1)
+    project_classifiers = re.findall(r'classifiers\s*=\s*\[([^\]]*)\]', setup_py, re.DOTALL)
+    project_classifiers = project_classifiers[0].replace('\n', '').replace(' ', '').replace('"', '').split(',')
+    project_packages = re.findall(r'packages\s*=\s*\[([^\]]*)\]', setup_py, re.DOTALL)
+    project_packages = project_packages[0].replace('\n', '').replace(' ', '').replace('"', '').split(',')
+    project_install = re.findall(r'install_requires\s*=\s*\[([^\]]*)\]', setup_py, re.DOTALL)
+
+    # Load the file "./{{cookiecutter.project_name}}/pyproject.toml" as the pyproject.toml template
+    script_dir = os.path.dirname(os.path.realpath(__file__))
+    template_file = os.path.join(script_dir, '{{cookiecutter.project_name}}', 'pyproject.toml')
+    with open(template_file, 'r') as f:
+        template = f.read()
+
+    # Replace placeholders in the template with the extracted data
+    template = template.replace('{{ cookiecutter.python_package_name }}', project_name)
+    template = template.replace('version = "0.1.0"', f'version = "{project_version}"')
+    template = template.replace('{{ cookiecutter.project_short_description }}', project_description)
+    template = template.replace('{{ cookiecutter.full_name }}', project_author)
+    template = template.replace('{{ cookiecutter.email }}', project_author_email)
+    template = template.replace('{{ cookiecutter.download_url }}', project_url)
+
+    setup_dirname = os.path.dirname(setup_python_path)
+    if os.path.exists(os.path.join(setup_dirname, 'README.md')):
```

Comment:
Eventually, all `README` files should be translated to either markdown or restructuredtext :upside_down_face:.

Reply:
Agreed. We could do this as a follow-up step.